analyzers/src/SonarAnalyzer.CSharp/Rules/CognitiveComplexity.cs

```diff
@@ -37,6 +38,21 @@ public sealed class CognitiveComplexity : CognitiveComplexityBase<SyntaxKind>
 
         protected override void Initialize(ParameterLoadingAnalysisContext context)
         {
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
+                {
+                    if (c.ContainingSymbol is IMethodSymbol)
+                    {
+                        CheckComplexity<CompilationUnitSyntax>(
+                            c,
+                            cu => cu,
```

Comment:
I had to look for the type to understa what `cu` actually is and understand the code. We try to avoid abreviations. Can please rename this to `compilationUnit`?